filament/backend/src/vulkan/VulkanBinder.cpp

```diff
@@ -37,8 +37,8 @@ static constexpr VkAllocationCallbacks* VKALLOC = nullptr;
 // TODO: Make VulkanBinder robust against a large number of descriptors.
 // There are several approaches we could take to deal with too many descriptors:
 // - Create another pool after hitting the limit of the first pool.
-// - Allow up to 3 uniform descriptors to bind simultaneously instead of just 1.
-static constexpr uint32_t MAX_NUM_DESCRIPTORS = 1000;
+// - Allow several uniform buffer descriptors to bind simultaneously instead of just one.
+static constexpr uint32_t MAX_DESCRIPTOR_SET_COUNT = 1500;
```

Comment:
Just curious, how many descriptors do we typically generate? I'm surprised we get anywhere near to 1000

Reply:
We go over 1000 now for some scenes because we never mutate descriptor sets; we create one for each unique combination of "bindings" across a given frame, where a binding is a single uniform buffer or sampler.  This strategy makes life easy because it avoids synchronization problems.

Since we sometimes run into the limit, we should consider switching to the strategy that is used in bgfx, which has one "uber" uniform buffer and doesn't need many descriptor sets.

Reply:
Cool, I'm curious because Metal has something similar called "argument buffers" which we're not currently using, but I'm considering switching over to in the future